frontend/src/stores/workspace.ts

```
@@ -0,0 +1,190 @@
+import { defineStore } from 'pinia';
+import type { PersistenceOptions } from 'pinia-plugin-persistedstate';
+import { ref, computed } from 'vue';
```

@BenBotros 


- [ ] Use centralized API endpoints and remove hardcoded URLs.
    - [ ] Use `api` import for API calls.
- [ ] fetch is get so replace fetchUnits by getUnits, let's use HTTP VERB get/post/path/delete
- [ ] Rewrite `fetchUnit` to fetch a single unit by ID.
- [ ] Use the `options` parameter in `fetchUnitResults` for pagination/sorting.
- [ ] try catch with empty catch is bad practice errors in `initFromPersisted` for better debugging.
    -> let's. think of a way of handling errors globally maybe ? 


- [ ] Remove console logs or replace with proper logging.
- [ ] Clean up redundant or obvious comments.
- [ ] Use separate loading states for different fetch actions.
     -> const units
     -> const unitsLoading
     -> const unitsErrors []

@guilbep maybe later in new issue https://github.com/EPFL-ENAC/co2-calculator/issues/118
- [ ] Ensure interfaces match backend responses and are documented.
    --> maybe later @guilbep (https://openapi-ts.dev/openapi-fetch/ + openapi-typescript)
- [ ] Add retry or fallback logic for failed fetches.
  -> todo later for @guilbep